Essentials/src/main/java/com/earth2me/essentials/commands/Commandvanish.java

```diff
@@ -40,6 +45,30 @@ void togglePlayer(final CommandSource sender, final User user, Boolean enabled)
         if (enabled) {
             user.sendMessage(tl("vanished"));
         }
+
+        if (ess.getSettings().isFakeMessageOnVanish()) {
+            final Player player = user.getBase();
+            String msg;
+            if (enabled) {
+                if (ess.getSettings().isCustomQuitMessage()) {
+                    msg = ess.getSettings().getCustomQuitMessage();
+                } else {
+                    msg = "&e{USERNAME} left the game";
+                }
+            } else {
+                if (ess.getSettings().isCustomJoinMessage()) {
+                    msg = ess.getSettings().getCustomJoinMessage();
+                } else {
+                    msg = "&e{USERNAME} joined the game";
```

Comment:
must be configurable, plugins may alter the join/leave message.

Reply:
That's how it was at https://github.com/EssentialsX/Essentials/pull/3881/commits/1d5b94f7fe4159d61e40968586d2f447a3cad119. Should I revert to the way it was at that point or is there a better way to make them configurable?